### DISCUSSION TITLE
Supabase real-time changes only emit on DELETE events.

### DISCUSSION THREAD

**Discussion by Halim567** [supabase/supabase#18558] [Questions]
Hi, I'm new to Supabase, and I'm having trouble with real-time changes. When I perform an insert or update, I do not receive any real-time events.

```
onMount(() => {
        const subscribe = supabase.channel('schema-db-changes')
            .on('postgres_changes', { event: '*', table: 'ChatUsers', schema: 'public'}, payload => {
                console.log(payload);
            }).subscribe();

        return () => subscribe.unsubscribe();
});
```
![Screenshot 2023-10-28 235656](https://github.com/supabase/supabase/assets/129165181/df4b4396-2127-4271-bb6c-83d3dbeeea4a)

**Comment by GaryAustin1** [comment]
Seems like you don't have a session (user jwt) set in realtime, IF your policy is really on ALL as you are just showing a label.  You have to meet select policy for everything but delete.
You could add `.subscribe(status=>console.log(status))` and see if you get any errors.
You can also use https://realtime.supabase.co/inspector/new to make sure your instance is set up properly.

**Reply by Halim567** [reply]
Hmmm, I still can't find the issue. I added `.subscribe(status => console.log(status))`, and it outputs 'SUBSCRIBED'. I also tried https://realtime.supabase.co/inspector/new, and it gives me:

`{"schema":"public","table":"ChatUsers","commit_timestamp":"2023-10-29T04:00:53.583Z","eventType":"INSERT","new":{"created_at":"2023-10-29T04:00:53.579047+00:00","dislikedby":null,"dislikes":0,"id":19,"likedby":null,"likes":0,"pesan":"test","update_at":"2023-10-29T04:00:53.579047+00:00","user_id":"7ab07f24-0749-4fb4-b42c-db70c202197e","username":"gogon"},"old":{},"errors":null}`.

**Replication**

![Screenshot 2023-10-29 110655](https://github.com/supabase/supabase/assets/129165181/7f0f542b-8f1b-48b6-9e6b-2fd4eed9cbf2)

Oh yeah, how do I set the session to realtime?

**Comment by bernhardw** [comment]
Hello @Halim567, same problem here... did you figure it out?

**Comment by typed-sigterm** [comment]
Same problem here and still looking for a solution...

**Reply by GaryAustin1** [reply]
This has always been you are not meeting select RLS for many different reasons (if you get deletes working and nothing else).
You have to figure why you don't meet select RLS on the table as it could be no user session when you subscribe to realtime, lost session later, just bad RLS on the table.   You provided no info so no one can even suggest where to look.
